features/22-email-verification/api.md

```diff
@@ -0,0 +1,111 @@
+# Verify by email
+The use case and requirement apply to verify-by-email and verify-by-sms.
+
+## Use cases
+* App may request the user to verify the email address/phone number upon signup.
+* The app need will send an email/SMS to the user with the email address/phone number provided when logging in. The app will want to customize the content of the email/SMS.
+* Both email/SMS verification supports verify by a verification code or by a verification link.
+	* For verification code, the user will type the verification code from the email address/SMS an enter it to the app to complete verification.
+	* For verification link, the user will be redirected to a webpage, which automatically send the code to the server to complete verification.
+* The App is able to check the verification status of the user using the SDK and cloud function so that it can display other info and/or disable features.
+* The developer may want to deny unverified user from accessing any server API except auth related ones.
+
+## Requirements
+* Server will have configuration options for:
```

Comment:
Another configuration I can think of is the complexity of verification code. We can start with a simple one, or a setting to allow a simple and a complicated one.

Reason is most of the time, for apps, it is pretty simple, could be 4 - 6 digits, for ease of input from SMS to apps.

For web, usually is a bit more complicated.

Reply:
Yes! I also think we should have different style of verification code something like `123456` for SMS and `352119F5C0B0` for email. Configurable of course.